doc/development/step_by_step_guide_to_writing_your_own_dialect.md

```diff
@@ -44,6 +46,32 @@ Create an empty unit test class for the dialect: `com.exasol.adapter.dialects.at
 
 Now that you have the skeleton of the dialect adapter, it is time to implement the specifics.
 
+## Creating the SQL Dialect Factory
+
+Each dialect is accompanied by factory that is responsible for instantiating that dialect. The [Java Service](https://docs.oracle.com/javase/8/docs/api/java/util/ServiceLoader.html) loader takes care of finding an loading the factory. It looks up the fully qualified class name of the dialect factories in the file [`com.exasol.adapter.dialects.SqlDialectFactory`](../../jdbc-adapter/virtualschema-jdbc-adapter/src/main/resources/META-INF/services/com.exasol.adapter.dialects.SqlDialectFactory).
+
+The factory itself is very simple. It only has two methods that you need to implement and that implementation is surprisingly simple. First the factory needs to be able to provide the dialect name since the JDBC adapter takes identifies dialect by name.
```

Comment:
```suggestion
The factory itself is very simple. It only has two methods that you need to implement and that implementation is surprisingly simple. First the factory needs to be able to provide the dialect name since the JDBC adapter identifies dialect by name.
```